test/functional/gcp_network_state.go

```diff
@@ -0,0 +1,203 @@
+package functional
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"cloud.google.com/go/compute/apiv1/computepb"
+	croGCP "github.com/integr8ly/cloud-resource-operator/pkg/providers/gcp"
+	"github.com/integr8ly/cloud-resource-operator/pkg/providers/gcp/gcpiface"
+	croResources "github.com/integr8ly/cloud-resource-operator/pkg/resources"
+	"github.com/integr8ly/integreatly-operator/test/common"
+	"google.golang.org/api/option"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utils "k8s.io/utils/pointer"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	defaultGcpServiceConnectionURI = "servicenetworking.googleapis.com"
+	defaultGcpServicesFormat       = "services/%s"
+	defaultIpRangePostfix          = "ip-range"
+	defaultGcpIdentifierLength     = 40
+	gcpTier                        = "production"
+	gcpAllowedCidrRanges           = []string{
+		"10.255.255.255/8",
+		"172.31.255.255/12",
+	}
+)
+
+// TestGCPNetworkState tests GCP cloud network components
+func TestGCPNetworkState(t common.TestingTB, testingCtx *common.TestingContext) {
+	ctx := context.Background()
+
+	// get the strategy map to get the GCP Subnet cidr block
+	strategyMap := &corev1.ConfigMap{}
+	err := testingCtx.Client.Get(ctx, types.NamespacedName{
+		Namespace: common.RHOAMOperatorNamespace,
+		Name:      croGCP.DefaultConfigMapName,
+	}, strategyMap)
+	if err != nil {
+		t.Fatal("could not get gcp strategy map", err)
+	}
+
+	strat, err := getStrategyForResource(strategyMap, networkResourceType, gcpTier)
+	if err != nil {
+		t.Skip("_network key does not exist in strategy configmap, skipping standalone vpc network test")
+	}
+
+	// get the cidr block from Strategy Map
+	expectedCidr, err := verifyAndGetCidrBlockFromGCPStrategyMap(strat)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	serviceAccountJson, err := getGCPCredentials(ctx, testingCtx.Client)
+	if err != nil {
+		t.Fatal("failed to retrieve gcp credentials %w", err)
```

Comment:
AFAIK, using the `%w` verb can only be used with `fmt.Errorf()`.
The same problem can be seen in most of the `t.Fatal()` invocations here.
Something like that should do:
`t.Fatalf("<some error message>: %s", err)`

Reply:
Updated